File: Kotlin/merge-strings-alternately.kt
class Solution {
    fun mergeAlternately(word1: String, word2: String): String {
        val m = word1.length
        val n = word2.length
        val merged = StringBuilder(m + n)
        var i = 0
        val minLen = minOf(m, n)
        while (i < minLen) {
            merged.append(word1[i])
            merged.append(word2[i++])
        }
        if (i < m) {
            merged.append(word1, i, m)
        }
        if (i < n) {
            merged.append(word2, i, n)
        }
        return merged.toString()
    }
}

File: C#/merge-strings-alternately.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        int i = 0, j = 0, k = 0;
        var sb = new char[word1.Length + word2.Length];
        while (i < word1.Length && j < word2.Length) {
            sb[k++] = word1[i++];
            sb[k++] = word2[j++];
        }
        while (i < word1.Length) {
            sb[k++] = word1[i++];
        }
        while (j < word2.Length) {
            sb[k++] = word2[j++];
        }
        return new string(sb);
    }
}